### ISSUE TITLE
IDirectoryContents doesn't know it's own name or physical path

### ISSUE THREAD

**Issue by Tratcher** [dotnet/runtime#36496]
GetDirectoryContents returns an IDirectoryContents that has an Exists property and an enumeration of files. What it's missing compared to IFileInfo is the Name and PhysicalPath properties.  I can't pass the IDirectoryContents around without wrapping it in my own structure that remembers the name I passed in to retrieve it, and there's no way to get the PhysicalPath except to inspect one of the items in the directory.

Fixing GetFileInfo to support directories would help too (https://github.com/aspnet/Home/issues/2537).

Found while app building for https://github.com/aspnet/AuthSamples/issues/28

**Comment by maryamariyan** [comment]
> Not yet. A quick GitHub search shows that there are indeed some usages of the interface so it would be best to keep the breaking changes to a minimum

Agreed. We have a high bar for breaking changes. In order to consider this change we would first need to come up with an API prosoal as explained in the guideline [api-review-process.md](https://github.com/dotnet/runtime/blob/master/docs/project/api-review-process.md).

@kshyju would you like to propose an API suggestion for this?
